### PR TITLE
Tickets/dm 9378

### PR DESCRIPTION
**Pull request by srp3rd**

### CODE REVIEW

File: python/lsst/ctrl/orca/WorkflowMonitor.py [hunk 1]
```diff
@@ -44,10 +44,10 @@ def addStatusListener(self, statusListener):
         log.debug("WorkflowMonitor:addStatusListener")
         self._statusListeners.append(statusListener)
 
-    def handleEvent(self, event):
-        """Act on an event request
+    def handleRequest(self, request):
+        """Act on a request
```
Comment (mxk62) [inline python/lsst/ctrl/orca/WorkflowMonitor.py:53]: I assume that currently it is just a stub as the `request` argument is not used. Even though I think it should be described in method's docstring.

File: python/lsst/ctrl/orca/StatusListener.py [hunk 2]
```diff
@@ -25,16 +25,17 @@
 
 
 class StatusListener(object):
-    """An interface for getting notified about changed in the status of a workflow
+    """An interface for getting notified about changed in the status of
+    a workflow
     """
 
     # initializer
```
Comment (mxk62) [inline python/lsst/ctrl/orca/StatusListener.py]: I find this comment really redundant.

File: python/lsst/ctrl/orca/StatusListener.py [hunk 3]
```diff
@@ -47,20 +48,15 @@ def workflowStarted(self, name):
...
+        Notes
+        -----
+        If a pipeline is waiting for an request, the listener should be
+        notified via workflowWaiting
         """
         return
 
     def workflowWaiting(self, name):
-        """Indicate that a workflow is waiting for an event to proceed.
+        """Indicate that a workflow is waiting for an request to proceed.
```
Comment (mxk62) [inline python/lsst/ctrl/orca/StatusListener.py:81]: Do I need to mention that docstrings of all methods in this class are not compliant with the LSST standards. ;-)

File: python/lsst/ctrl/orca/ServiceHandler.py [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+                self.parent.stopProduction(level)
+            except Exception as error:
+                self.send_response(422)
+                self.end_headers()
+                self.writeError("Unprocessable entity", "Error in syntax of message")
+            return
+        self.send_response(400)
+        self.writeError("Bad Request", "Request is unsupported")
+        self.end_headers()
+
```
Comment (mxk62) [inline python/lsst/ctrl/orca/ServiceHandler.py:78]: I have two quetions here.

Firstly, judging from `return` inside the `if` statement, you don't want lines 75-77 to be executed if the condition is fulfilled. Then why didn't you use `if-else`?

Secondly, you've packed a lot to `try-except`. Am I correctly assuming that you just don't care why the code fails and you are just interested to know whether it failed for some reason or not?

File: python/lsst/ctrl/orca/StatusListener.py (hunk 3 above)
Comment (srp3rd) [reply python/lsst/ctrl/orca/StatusListener.py:81]: that class  predates most of those standards by a number of years.  :-)   I'll fix this here;  the whole package is due for a scrub to clean all this up to be sure it's all up to standard for autopep8.

File: python/lsst/ctrl/orca/ServiceHandler.py (hunk 4 above)
Comment (srp3rd) [reply python/lsst/ctrl/orca/ServiceHandler.py:78]: For the first point, the `return` makes the `else` clause unnecessary.   

Second point, yes.  I could get very detailed in the message section of `writeError` for each individual case, but the request message is simple enough I just opt-ed to have a general response than something more specific.

File: python/lsst/ctrl/orca/WorkflowMonitor.py (hunk 1 above)
Comment (srp3rd) [reply python/lsst/ctrl/orca/WorkflowMonitor.py:53]: Added

File: python/lsst/ctrl/orca/StatusListener.py (hunk 2 above)
Comment (srp3rd) [reply python/lsst/ctrl/orca/StatusListener.py]: Changed

**Pull request merged**